Extension/src/Debugger/configurationProvider.ts

```diff
@@ -36,29 +39,59 @@ abstract class CppConfigurationProvider implements vscode.DebugConfigurationProv
             return undefined;
         }
 
-        // Help WSL users with using the correct bash.exe
-        if (os.platform() === 'win32' && config.pipeTransport && config.pipeTransport.pipeProgram) {
-            const pipeProgramStr: string = config.pipeTransport.pipeProgram;
-
-            if (pipeProgramStr.indexOf("sysnative") >= 0 && process.arch === 'x64') {
-                // User has sysnative but is running VSCode 64 bit. Should be using System32
-                config.pipeTransport.pipeProgram = pipeProgramStr.replace("sysnative", "System32");
-                getOutputChannelLogger().appendLine(`WARNING: 64-bit VSCode should use System32 for the directory for pipeProgram.`);
-                getOutputChannelLogger().appendLine(`pipeProgram has been modified to be: ${config.pipeTransport.pipeProgram}`);
-                showOutputChannel();
-            } else if (pipeProgramStr.indexOf("System32") >= 0 && process.arch === 'ia32') {
-                // User has System32 but is running VSCode 32 bit. Should be using sysnative
-                config.pipeTransport.pipeProgram = pipeProgramStr.replace("System32", "sysnative");
-                getOutputChannelLogger().appendLine(`WARNING: 32-bit VSCode should use sysnative for the directory for pipeProgram.`);
-                getOutputChannelLogger().appendLine(`pipeProgram has been modified to be: ${config.pipeTransport.pipeProgram}`);
-                showOutputChannel();
+        const winDir = process.env.WINDIR? process.env.WINDIR.toLowerCase() : null;
```

Comment:
I think you want to check for `if (os.platform() === win32)` before you check for the `WINDIR` environment variable.